[CS-3864] Screen for initial login/unlock pin use

oidoug opened the pull request:
### Description

Adds UI for UnlockScreen template to be used as the initial login/unlock screen.

- [x] Completes #(CS-3864)

### Checklist

- [x] All UI changes have been tested on a small device

### Screenshots

https://user-images.githubusercontent.com/129619/170285402-5c9c7633-e7a6-4a1d-b7b9-cb2ec6796574.mov

https://user-images.githubusercontent.com/129619/170285749-0dd4ac83-668b-4e69-b237-1ff082261fb2.mp4

Commit:
Deriving Unlock Screen from PinScreen

Commit:
Adds UnlockScreen template to be used as the initial login screen.

Commit:
Improved bottom position for biometric toggle.

Commit:
Fix linting and usememo usage

danibonilha commented on cardstack/src/screens/UnlockScreen/UnlockScreen.tsx:
```diff
@@ -0,0 +1,157 @@
+
+import { useAppVersion } from '@rainbow-me/hooks';
+
+import cardwalletLogo from '../../assets/cardwalletLogoIntro.png';
```
I believe we already have this component on the WelcomeScreen and don't need to add a new asset

danibonilha commented on cardstack/src/screens/UnlockScreen/UnlockScreen.tsx:
```diff
@@ -0,0 +1,157 @@
+  return (
+    <>
+      <StatusBar barStyle={statusBarStyle} />
+      <NavigationStackHeader
+        canGoBack={true}
+        backgroundColor={colorStyleVariants.backgroundColor[variant]}
+      />
```
Not sure this screen should have an navigation header, and if it does the goBack option shouldn't be enable, bc will it go back to which place ?

oidoug replied:
Left it for testing, I'll remove it.

danibonilha commented on cardstack/src/screens/UnlockScreen/UnlockScreen.tsx:
```diff
@@ -0,0 +1,157 @@
+  const appVersion = useAppVersion();
+  const [inputPin, setInputPin] = useState('');
+
+  useEffect(() => {
+    Device.isAndroid && NativeModules?.AndroidKeyboardAdjust.setAdjustPan();
+  }, []);
+
+  const feedbackProps = useMemo(
+    () =>
+      isValidPin ? feedbackStatusProps.success : feedbackStatusProps.error,
+    []
+  );
+
+  const statusBarStyle = useMemo(
+    () => (variant === 'dark' ? 'light-content' : 'dark-content'),
+    []
+  );
```
lot's of dupe logic from pinScreen,  do we have a success state on this screen ? Does it make sense to copy the feedbackStatusProps ?

oidoug replied:
You're right, sorry, I'll clean this up.

danibonilha commented on cardstack/src/screens/UnlockScreen/UnlockScreen.tsx:
```diff
@@ -0,0 +1,157 @@
+                    </Text>
+                  </Container>
+                )}
+                {biometryAvailable && (
+                  <Container paddingTop={8} style={biometryBottomPaddingStyle}>
+                    <BiometricSwitch variant={variant} />
+                  </Container>
+                )}
+              </Container>
+              <Button marginVertical={6}>{strings.login.button}</Button>
```
The biometric switcher and the Log in button should be closer together

oidoug replied:
If I leave the switcher closer to the button, on a bigger screen, it will either be below the keyboard when it's open, or the whole bottom section needs to be higher up or more spaced between. Wdyt is better?

oidoug replied:
I'm trying a different approach, evenly distributing elements per section. What do you think?

https://user-images.githubusercontent.com/129619/170356730-1441a752-fff2-4dc3-a2ce-f675c3c26cf6.mov

oidoug replied:
Adjusted padding on logo:

<img width="400" alt="Screen Shot 2022-05-25 at 17 07 23" src="https://user-images.githubusercontent.com/129619/170358447-99f12f1b-afcd-4f4f-92ae-e7da8ff0880c.png">

danibonilha replied:
I believe the top of the logo should have a lower starting point on iPhone 13 Pro Max if we were to flex this design it would look something like this:  
<img width="200" alt="image" src="https://user-images.githubusercontent.com/20520102/170382605-8df14fb7-6dcd-44db-8db7-9f122c363346.png">

This way we can avoid setting fixed paddings
Maybe to fit even better we could use a smaller icon size on smaller devices, but not a requirement.

oidoug replied:
Ok! Done it and it's looking good :)

<img width="400" alt="Screen Shot 2022-05-26 at 09 37 36" src="https://user-images.githubusercontent.com/129619/170489287-bc4fda01-435d-40e6-bba5-e02bbd7f4a77.png">

danibonilha commented on cardstack/src/screens/UnlockScreen/UnlockScreen.tsx:
```diff
@@ -0,0 +1,157 @@
+              alignItems="center"
+            >
+              <Container width="100%" alignItems="center">
+                <PinInput
+                  autoFocus={false}
+                  variant={variant}
+                  value={inputPin}
+                  onChangeText={setInputPin}
+                />
```
It's missing the PIN code title

oidoug replied:
oh, sorry, missed that.

Commit:
Improves logo and UI positioniong in smaller screens

Commit:
Extracted CardwalletLogo from WelcomeScreen to reuse in Unlock screen;
Removed now unneded images;
Improved UI positioning;

Commit:
Adds pin title to PinInput;
Refactors error and strings in UnlockScreen.

Commit:
Removes unused import

Commit:
Relayout of containers, more evenly distribruted.

Commit:
Removing log

Commit:
Moves container props to CardwalletLogo comp;
Removes unused import.

danibonilha commented on cardstack/src/screens/UnlockScreen/UnlockScreen.tsx:
```diff
@@ -0,0 +1,126 @@
+  const { isSmallPhone } = useDimensions();
+  const { biometryAvailable } = useBiometricSwitch();
+  const appVersion = useAppVersion();
+  const [inputPin, setInputPin] = useState('');
+
+  useEffect(() => {
+    Device.isAndroid && NativeModules?.AndroidKeyboardAdjust.setAdjustPan();
+  }, []);
+
+  const statusBarStyle = useMemo(
+    () => (variant === 'dark' ? 'light-content' : 'dark-content'),
+    []
+  );
```
This could be shared between PIN and Unlock

oidoug replied:
Maybe put in a specific hook for setting input? I'll do this refactoring in my upcoming logic PR.

Commit:
Improves layout for UnlockScreen;
Adds size variant to CardwalletLogo comp;
Updates WelcomeScreen to use logo variant.

danibonilha reviewed (approved):
LGTM!
